src/open_inwoner/openproducten/tests/test_product_type_importer.py

```diff
@@ -0,0 +1,564 @@
+        ProductLink.objects.create(product=product)
+
+        importer = ProductTypeImporter(self.client)
+
+        importer._delete_non_updated_objects()
+        self.assertEqual(importer.deleted_count, 0)
+
+    def test_complete_import_with_new_objects(self):
+        related_product_type = create_product_type(uuid4())
+        product_type = create_complete_product_type("test2")
+        product_type.related_product_types.append(related_product_type.id)
+
+        self.client.fetch_producttypes_no_cache.return_value = [
+            product_type,
+            related_product_type,
+        ]
+        importer = ProductTypeImporter(self.client)
+        created, updated, deleted = importer.import_producttypes()
+
+        self.assertEqual(pdc_models.Product.objects.count(), 2)
+        self.assertEqual(pdc_models.ProductCondition.objects.count(), 1)
+        self.assertEqual(pdc_models.Tag.objects.count(), 1)
+        self.assertEqual(pdc_models.TagType.objects.count(), 1)
+        self.assertEqual(pdc_models.ProductLink.objects.count(), 1)
+        self.assertEqual(Price.objects.count(), 1)
+        self.assertEqual(PriceOption.objects.count(), 1)
+        self.assertEqual(pdc_models.Question.objects.count(), 1)
+
+        all_objects = get_all_product_type_objects()
+
+        self.assertEqual(updated, [])
+        self.assertEqual(deleted, 0)
+        self.assertEqual(
+            sorted([obj.open_producten_uuid for obj in created]),
+            sorted([obj.open_producten_uuid for obj in all_objects]),
+        )
+
+    def test_complete_import_with_existing_objects(self):
+        related_product_type = create_product_type(uuid4())
+        product_type = create_complete_product_type("test2")
+        product_type.related_product_types.append(related_product_type.id)
+
+        self.client.fetch_producttypes_no_cache.return_value = [
+            product_type,
+            related_product_type,
+        ]
+        importer = ProductTypeImporter(self.client)
+        importer.import_producttypes()
+
+        importer = ProductTypeImporter(self.client)
+        created, updated, deleted = importer.import_producttypes()
+
+        self.assertEqual(pdc_models.Product.objects.count(), 2)
+        self.assertEqual(pdc_models.ProductCondition.objects.count(), 1)
+        self.assertEqual(pdc_models.Tag.objects.count(), 1)
+        self.assertEqual(pdc_models.TagType.objects.count(), 1)
+        self.assertEqual(pdc_models.ProductLink.objects.count(), 1)
+        self.assertEqual(Price.objects.count(), 1)
+        self.assertEqual(PriceOption.objects.count(), 1)
+        self.assertEqual(pdc_models.Question.objects.count(), 1)
+
+        all_objects = get_all_product_type_objects()
+
+        self.assertEqual(created, [])
+        self.assertEqual(deleted, 0)
+        self.assertEqual(
+            sorted([obj.open_producten_uuid for obj in updated]),
+            sorted([obj.open_producten_uuid for obj in all_objects]),
+        )
```

Comment:
Als de volgorde er niet toe doet zijn sets leesbaarder:

```suggestion
        self.assertEqual(
            {obj.open_producten_uuid for obj in updated},
            {obj.open_producten_uuid for obj in all_objects},
        )
```

Reply:
(ook elders)